src/dist/replication/lib/replica_restore.cpp

```diff
@@ -444,27 +445,23 @@ dsn::error_code replica::restore_checkpoint()
 
     if (err == dsn::ERR_OK) {
         err = download_checkpoint(restore_req, remote_chkpt_dir, restore_dir);
-        if (err != ERR_OK) {
-            if (_restore_status == ERR_CORRUPTION) {
-                if (skip_bad_partition) {
-                    err = skip_restore_partition(restore_dir);
-                } else {
-                    tell_meta_to_restore_rollback();
-                    return ERR_CORRUPTION;
-                }
-            }
-        }
-    } else { // find valid checkpoint failed
-        if (err == ERR_OBJECT_NOT_FOUND) {
+        if (err != ERR_OK && _restore_status == ERR_CORRUPTION) {
```

Comment:
If `download_checkpoint` return error, we don't do any error handling?

Reply:
If `_restore_status == ERR_CORRUPTION`, then stop restore, or it will retry to restore.

Reply:
ok. I have another question. if `_restore_status = ERR_CORRUPTION`, will `download_checkpoint` return error_ok?

Reply:
No, `download_checkpoint` should not be `ERR_OK` if  `_restore_status != ERR_OK`

Reply:
ok, I suggest that the condition `if (err != ERR_OK && _restore_status == ERR_CORRUPTION) ` can be simplified as `if (_restore_status == ERR_CORRUPTION)`

Reply:
done